fsm_node:
  fsm:
    auto_pilot_en: true
    safe_auto_pilot: true
    click_goal_en: false
    odom_goal_en: false
    #    odom_goal_topic: "/odom_goal"
    #    click_goal_topic: "/planning/click_goal"
    click_goal_topic: "/goal"
    path_goal_en: false
    path_goal_topic: "/planning/path_goal"
    replan_rate: 10
    auto_pilot_speed: 5.0
    auto_pilot_yaw_speed: 1.0
    click_height: 1.5
    # if use the yaw ange in /goal as the goal yaw direction
    click_yaw_en: false
    #    cmd_topic: "/planning/pos_cmd"
    cmd_topic: "/quad_1/planning/pos_cmd"
    mpc_cmd_topic: "/planning_cmd/mpc"
    # if [true], the local goal vel will be set to vel_max heading to goal.
    # if [false], the local goal vel will be set to zero
    goal_vel_en: false
    goal_yaw_en: true


  svc_planner:
    print_log: false
    frontend_in_known_free: true
    visual_process: false
    use_fov_cut: false
    # 走廊生成的上限Bounding Box
    corridor_bound_dis: 1.2
    corridor_line_max_length: 1.5
    iris_iter_num: 3
    back_traj_dis_acc: 4.0
    obs_skip_num: 0
    # Replan时往前看多少秒
    replan_forward_dt: 0.1
    planning_horizon: 10
    receding_dis: 1.0
    # [WARN] the robot r should equal to decaymap resolution[0] * infalate dis.
    robot_r: 0.25
    yaw_dot_max: 1.2
    # Yaw mode: 1 heading to velocity, 2 heading to goal
    yaw_mode: 1


  traj_opt:
    exp_traj:
      energy_cost_type: 4
      block_energy_cost: false
      opt_accuracy: 1.0e-5
      scale_factor: 1.0
      integral_reso: 10
      smooth_eps: 0.01
      max_vel: 1.0
      max_acc: 1.0
      max_jerk: 60.0
      penna_t: 30.0
      penna_pos: 1.0e+6
      penna_vel: 5.0e+5
      penna_acc: 5.0e+5
      penna_jerk: 1.0e+5 #1.0e+8 #
      penna_attract: 1.0e+5

    backup_traj:
      energy_cost_type: 4
      block_energy_cost: false
      opt_accuracy: 1.0e-5
      scale_factor: 1.0
      integral_reso: 10
      smooth_eps: 0.01
      max_vel: 1.5
      max_acc: 2.0
      max_jerk: 50.0
      penna_t: 120.0
      penna_pos: 1.0e+8
      penna_vel: 1.0e+6
      penna_acc: 1.0e+6
      penna_jerk: -1


  astar:
    map_voxel_num: [ 200, 200, 51 ]
    visual_process: false
    allow_diag: false
    # 0 DIAG; 1 MANHATTAN; 2 EUCLIDEAN
    heu_type: 2

  rog_map:
    safe_margin: 0.0
    resolution: 0.05
    inflation_resolution: 0.15
    inflation_step: 2
    map_size: [ 30,30,5 ]

    # Virtual ceil and ground
    virtual_ceil_height: 999
    virtual_ground_height: -999

    map_sliding:
      enable: true
      # The minimum distance [m] to slide the map.
      threshold: 0.05
      # If the map rolling is disable, the map origin [m] should be set.
      fix_map_origin: [ 0,0,0 ]

    frontier_extraction_en: false

    # If [enbale = true], the ROG-Map will actively take ros topic as input.
    #  else user should call function [updateMap] to update the map.
    ros_callback:
      enable: true
      #      cloud_topic: "/quad0_pcl_render_node/cloud"
      #      odom_topic: "/quad_0/lidar_slam/odom"
      cloud_topic: "/cloud_registered"
      odom_topic: "/lidar_slam/odom"
      odom_timeout: 2.0

    visulization:
      enbale: true
      # The rate of visualization [Hz], if set to 0, the visualization will not be triggered by frame.
      time_rate: 10
      # The frame rate of visualization [Hz], if set to 0, the visualization will not be triggered by frame.
      frame_rate: 0
      # visualization range along the odom [m].
      range: [ 4,4,2 ]
      frame_id: "world"
      pub_unknown_map_en: true




    # Intensity filter for input point cloud. If the point intensity is less than this value,
    # it will be ignored (to avoid noise like dusk).
    intensity_thresh: -1
    # The temporal downsample rate for input point cloud.
    #  e.g. if set to k, the point cloud will be downsampled to 1/k.
    point_filt_num: 3

    # Probabilistc update
    raycasting:
      # if disable, the map will only maintain occupied information, and all other grid
      # will be considered as unknown.
      enable: true
      batch_update_size: 1
      local_update_box: [ 5, 5, 3 ]
      # The range of raycasting [m].
      ray_range: [ 0.5, 5 ]

      p_min: 0.12
      p_miss: 0.48
      p_free: 0.49
      p_occ: 0.8
      p_hit: 0.8
      p_max: 0.98
      # The threshold of a non-occupied inf map grid to be considered as free. [0.0-1.0]
      # which means for a gird contains [u] unknown grids and [f] known free grids.
      # if [f/(f+u) > inf_map_known_free_thresh], the grid will be considered as free.
      #  e.g. this thresh is set to 1.0, it measn a inf map grid is considered as free only if all grids in it are known free.
      inf_map_known_free_thresh: 1.0